security/fma-interop-tx-handling.md

```diff
@@ -0,0 +1,168 @@
+
+**Check Every Message Exhaustively**
+- In this model, every Executing Message is checked constantly, at a maximum rate, and each message is checked as close to block-building as possible.
+- The compute cost to check every message during block building adds time to *every* transaction, blowing out our ability
+to build a block in under 2s.
+- The validity of the included Executing Messages would be *as correct* as possible. However! Even *after* the block is built, the data being relied upon (cross-unsafe data) could change on the Initiating Chain if they suffer a reorg.
+So while this policy is *most correct*, it is not *totally correct*
+
+**Never Check Any Messages**
+- In this model, we optimize only for not taking any additional compute tasks, instead just trusting every message.
+- Naturally, there is no impact into block building or any other process, BUT...
+- Blocks would easily become invalid, because an attacker could submit Invalid Messages, even just one per 2s, and prevent the Sequencer from ever building a valid block.
+
+So, no matter what solution we pick, we deal with *some* amount of uncertainty and take *some* amount of additional compute load.
+
+## The Solution Design
+
+The [Interop Topology and Tx Flow for Interop Chains Design Doc](https://github.com/ethereum-optimism/design-docs/pull/218)
+describes the solution design we plan on going with:
+
+- All Executing Message are checked once at `proxyd` ingress.
+- All Executing Message are checked once at Node Mempool ingress (not counting Sequencer).
+- All Executing Message in Node Mempools are Batched at checked on a regular interval.
+- If an Executing Message is ever Invalid, it is discarded and not retried.
+- *No* Checks are done at Block Building time.
+
+This FMA describes the potential negative consequences of this design. We have selected this design because it maximizes
+the opportunities for Invalid Messages to be caught and discarded, while also leaving the block building hot-path from
+having to take on new compute.
+
+# FMs:
+
+## FM1: Checks Fail to Catch an Invalid Executing Message
+- Description
+    - This overlaps Supervisor FMA's FM2a, where an invalid message is included in a built block.
+    - An interop transaction is checked mulitiple times (Proxyd, Sentry Node, Sequencer), and each check must fail
+    in order for the transaction to make it all the way to block building.
+    - The most likely cause for all filter layers to fail is a bug in the Supervisor.
+    - Individual layers may fail to filter a transaction if it decides the transaction doesn't need to be
+    checked for Interop Validity. For example, if it appears to have no Access List, it does not need to be checked.
+    - Recurring filters (ones which regularly re-validate Transactions in a Mempool) may fail to filter
+    a Transaction if the frequency of the check is too low.
+- Risk Assessment
```

Comment:
Recommended action items:
- Monitor for invalid messages
- Monitor for critical messages

Reply:
what is a critical message?